captum/optim/_param/image/images.py

```diff
@@ -1,538 +1,446 @@
-            nn.Parameter tensor.
-        batch (int): The number of channels to use when creating the
-            nn.Parameter tensor, or stacking init images.
-        parameterization (ImageParameterization, optional): An image parameterization
-            class.
-        squash_func (Callable[[torch.Tensor], torch.Tensor]], optional): The squash
-            function to use after color recorrelation. A funtion or lambda function.
-        decorrelation_module (nn.Module, optional): A ToRGB instance.
-        decorrelate_init (bool, optional): Whether or not to apply color decorrelation
-            to the init tensor input.
-    """
-
-    def __init__(
-        self,
-        size: Tuple[int, int] = None,
-        channels: int = 3,
-        batch: int = 1,
-        init: Optional[torch.Tensor] = None,
-        parameterization: ImageParameterization = FFTImage,
-        squash_func: Optional[Callable[[torch.Tensor], torch.Tensor]] = None,
-        decorrelation_module: Optional[nn.Module] = ToRGB(transform="klt"),
-        decorrelate_init: bool = True,
-    ) -> None:
-        super().__init__()
-        self.decorrelate = decorrelation_module
-        if init is not None:
-            assert init.dim() == 3 or init.dim() == 4
-            if decorrelate_init:
-                assert self.decorrelate is not None
-                init = (
-                    init.refine_names("B", "C", "H", "W")
-                    if init.dim() == 4
-                    else init.refine_names("C", "H", "W")
-                )
-                init = self.decorrelate(init, inverse=True).rename(None)
-            if squash_func is None:
-                squash_func: Callable[[torch.Tensor], torch.Tensor] = lambda x: x.clamp(
-                    0, 1
-                )
-        else:
-            if squash_func is None:
-                squash_func: Callable[
-                    [torch.Tensor], torch.Tensor
-                ] = lambda x: torch.sigmoid(x)
-        self.squash_func = squash_func
-        self.parameterization = parameterization(
-            size=size, channels=channels, batch=batch, init=init
-        )
-
-    def forward(self) -> torch.Tensor:
-        image = self.parameterization()
-        if self.decorrelate is not None:
-            image = self.decorrelate(image)
-        image = image.rename(None)  # TODO: the world is not yet ready
-        return CudaImageTensor(self.squash_func(image))
+from copy import deepcopy
+from typing import Callable, List, Optional, Tuple, Union
+
+import matplotlib.pyplot as plt
+import numpy as np
+import requests
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+try:
+    from PIL import Image
+except (ImportError, AssertionError):
+    print("The Pillow/PIL library is required to use Captum's Optim library")
+
+from captum.optim._param.image.transform import SymmetricPadding, ToRGB
+
+
+class ImageTensor(torch.Tensor):
```

Comment:
So, I was looking at the code, and it doesn't seem like the current ImageTensor works with CUDA tensors:

```
class ImageTensor(torch.Tensor):
     ...

x = torch.ones(1,3,4,4).cuda()
x_out = ImageTensor(x)

# TypeError: expected CPU (got CUDA)
```

Reply:
For `ImageTensor`, the way to set cuda is
```
x = torch.ones(1, 3, 4, 4)
x_out = ImageTensor(x).cuda()
```
The same applies for other Tensors like `FloatTensor`.
So doing
```
x = torch.ones(1,3,4,4).cuda()
x_out = torch.FloatTensor(x)
```
also throws the same TypeError.

But let me know if we should have different behavior for ImageTensor

Reply:
If you look at `NaturalImage`'s forward function, the parameterization will either have to always be on the CPU, or we'll have to do a conversion to and from the CPU when using a GPU. Conversions between GPUs and the CPU slow things down, and thus it's better to avoid them as much as possible.

```
    def forward(self) -> torch.Tensor:
        image = self.parameterization()
        if self.decorrelate is not None:
            image = self.decorrelate(image)
        image = image.rename(None)  # TODO: the world is not yet ready
        return ImageTensor(self.squash_func(image))
```

Reply:
@greentfrapp This issue should be resolved now!

Reply:
Though you should probably add a test to verify that CUDA tensors work with `ImageTensor` now.

Reply:
Added cuda test!